src/oc_erchef/apps/chef_objects/src/chef_password.erl

```diff
@@ -99,9 +102,17 @@ verify(Password, {HashedPass, Salt, _}) ->
     end.
 
 sha1(Salt, Password, osc) ->
-    hexstring(crypto:hash(sha, ["--", Salt, "--", Password, "--"]));
+    hexstring(safe_crypto(["--", Salt, "--", Password, "--"]));
 sha1(Salt, Password, ec) ->
-    hexstring(crypto:hash(sha, [Salt, "--", Password, "--"])).
+    hexstring(safe_crypto([Salt, "--", Password, "--"])).
+
+%% Eat any exceptions so secure data doesn't escape
+safe_crypto(Data) ->
+    try crypto:hash(sha, Data) of
+        Digest -> Digest
+    catch
+        _ -> <<"crypto:hash failure">>
```

Comment:
Wonder if it's worth mocking crypto:hash and having it return an exception to cover this side of the branch?